theano/tensor/nnet/corr_gemm.c

```diff
@@ -172,18 +172,19 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
     }
 
     // Create temporary columns
-    npy_intp col_dim[2];
-    col_dim[0] = (npy_intp)(nChannels * kW * kH);
-    col_dim[1] = (npy_intp)(topHeight * topWidth);
-    PyArrayObject* col = (PyArrayObject*)PyArray_EMPTY(2,
+    npy_intp col_dim[3];
+    col_dim[0] = (npy_intp)(batchSize);
```

Comment:
This is ok for testing, but use too much memory. You only need the size min(batch_size, number_of_thread).
I think the max number of thread can be know by a call to openmp. This will need adjustmane in the usage of col.